server/src/main/java/org/opensearch/search/slice/SliceBuilder.java

```diff
@@ -227,20 +227,17 @@ public Query toFilter(ClusterService clusterService, ShardSearchRequest request,
 
         int shardId = request.shardId().id();
         int numShards = context.getIndexSettings().getNumberOfShards();
-        if (minNodeVersion.onOrAfter(LegacyESVersion.V_6_4_0) && (request.preference() != null || request.indexRoutings().length > 0)) {
+        if ((request.preference() != null || request.indexRoutings().length > 0)) {
```

Comment:
Super minor, `((` is redundant here, just `(` is enough

Reply:
will clean up